lib/action_view/component/content_for.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentFor
+      def content_for(attr, content = nil, options = {}, &block)
+        unless respond_to?(attr)
+          raise StandardError.new "Called content_for with unknown attribute '#{attr}'"
+        end
+
+        if block_given?
+          options = content if content
+          content = view_context.capture(&block)
+        end
+
+        existing_content = send(attr)
+        if existing_content && !options[:flush]
+          content = existing_content + content
+        end
+
+        set_attribute(attr, content)
+        nil
+      end
+
+      private
+
+      def set_attribute(attr, value)
```

Comment:
If we go with the `content_areas` registration approach above, I think we should consider simplifying this to just use `attr_readers`.